src/main/java/menion/android/whereyougo/gui/activity/XmlSettingsActivity.java

```diff
@@ -1,60 +1,48 @@
 package menion.android.whereyougo.gui.activity;
 
 import android.app.Activity;
-import android.app.Dialog;
-import android.content.DialogInterface;
 import android.content.Intent;
 import android.content.SharedPreferences;
-import android.content.res.Configuration;
 import android.media.RingtoneManager;
 import android.net.Uri;
-import android.os.Build;
 import android.os.Bundle;
-import android.preference.CheckBoxPreference;
-import android.preference.Preference;
-import android.preference.PreferenceActivity;
-import android.preference.PreferenceManager;
-import android.preference.PreferenceScreen;
+import android.view.WindowManager;
+import android.widget.TextView;
 
-import java.io.File;
-import java.util.ArrayList;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.Fragment;
+import androidx.preference.Preference;
+import androidx.preference.PreferenceFragmentCompat;
+import androidx.preference.PreferenceManager;
 
-import ar.com.daidalos.afiledialog.FileChooserDialog;
-import menion.android.whereyougo.MainApplication;
 import menion.android.whereyougo.R;
-import menion.android.whereyougo.gui.utils.UtilsGUI;
-import menion.android.whereyougo.preferences.Locale;
 import menion.android.whereyougo.preferences.PreferenceValues;
 import menion.android.whereyougo.preferences.Preferences;
-import menion.android.whereyougo.preferences.PreviewPreference;
-import menion.android.whereyougo.utils.A;
-import menion.android.whereyougo.utils.FileSystem;
 import menion.android.whereyougo.utils.Logger;
-import menion.android.whereyougo.utils.ManagerNotify;
-import menion.android.whereyougo.utils.StringToken;
 import menion.android.whereyougo.utils.Utils;
 
+import static menion.android.whereyougo.gui.extension.activity.CustomActivity.setLocale;
 
-public class XmlSettingsActivity extends PreferenceActivity
-        implements SharedPreferences.OnSharedPreferenceChangeListener,
-        Preference.OnPreferenceClickListener {
 
-    private static final String TAG = "XmlSettingsActivity";
-
-    private boolean needRestart;
+public class XmlSettingsActivity
+    extends AppCompatActivity
+    implements PreferenceFragmentCompat.OnPreferenceStartFragmentCallback {
 
-    private static String getKey(final int prefKeyId) {
-        return Locale.getString(prefKeyId);
-    }
+    private static final String TAG = "XmlSettingsActivity";
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setContentView(R.layout.layout_settings);
         setTitle(R.string.settings);
+        ((TextView) findViewById(R.id.title_text)).setText(R.string.settings);
 
-        needRestart = false;
+        // Set language
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
+        String lang = sharedPreferences.getString(getString(R.string.pref_KEY_S_LANGUAGE), "");
+        setLocale(this, lang);
```

Comment:
What format does the language preference string have? According to the deleted part in `MainApplication.java` it may be either `xx` or `xx_yy`. In the second case we need to use `setLocale(this, lang, country)` IIRC.

Reply:
I havn't added the country becuase I was not sure if I add it, this may lead to bugs but so may this implementation do. Thus would you prefer if I add the country?

Reply:
Is it possible to leave the old language implementation as is?
At lease for pt we have also country languages ("pt-rBR","pt-rPT").

Reply:
Changing the language de<=>en worked in my test.
Did not tested it deeper.
Maybe we should create an issue to track it (and mybe do it as in c:geo).